experimental/sql/dialects/rel_ssa.py

```diff
@@ -0,0 +1,271 @@
+
+  schema = ParameterDef(ArrayOfConstraint(DataType))
+
+  @staticmethod
+  @builder
+  def get(types: list[DataType]) -> 'Bag':
+    return Bag([ArrayAttr.from_list(types)])  #type: ignore
+
+
+class Expression(Operation):
+  """
+  Interface class for all Expressions, i.e., operations that work on single
+  values.
+  """
+  ...
+
+
+@irdl_op_definition
+class Column(Expression):
+  """
+  References a specific column with name `col_name` and returns its value.
+
+  Example:
+
+  '''
+  %0 : rel_ssa.int32 = rel_ssa.column() ["col_name" = "id"]
+  '''
+  """
+  name = "rel_ssa.column"
+
+  col_name = AttributeDef(StringAttr)
+  result = ResultDef(DataType)
+
+
+@irdl_op_definition
+class Compare(Expression):
+  """
+  Returns `left` 'comparator' `right`.
+
+  Example:
+
+  '''
+  ...
+  %0 : !rel_ssa.bool = rel_ssa.compare(%a : !rel_ssa.int32, %b : !rel_ssa.int32) ["comparator" = "="]
+  '''
+  """
+  name = "rel_ssa.compare"
+
+  left = OperandDef(DataType)
+  right = OperandDef(DataType)
+  comparator = AttributeDef(StringAttr)
+
+  result = ResultDef(Boolean)
+
+  @staticmethod
+  @builder
+  def get(left: Operation, right: Operation,
+          comparator: StringAttr) -> 'Compare':
+    return Compare.build(operands=[left, right],
+                         attributes={"comparator": comparator},
+                         result_types=[Boolean])
+
+
+@irdl_op_definition
+class Yield(Expression):
+  """
+  Bridges the gap from expressions back to operators by yielding values.
+
+  Example:
+  '''
+  rel_ssa.yield(%0 : !rel_ssa.bool)
+  '''
+  """
+  name = "rel_ssa.yield"
+
+  ops = VarOperandDef(AnyAttr())
+
+  @staticmethod
+  @builder
+  def get(ops: list[Operation]) -> 'Yield':
+    return Yield.build(operands=[ops])
+
+
+@irdl_op_definition
+class Literal(Expression):
+  """
+  Creates a literal with value `value`.
+
+  Example:
+
+  '''
+  %0 : !rel_ssa.int32 = rel_ssa.literal() ["value" = 5 : i32]
+  '''
+  """
+  name = "rel_ssa.literal"
+
+  # TODO: change IntegerAttr s.t. it can have type !rel.int32
+  value = AttributeDef(AnyAttr())
+  result = ResultDef(DataType)
+
+  @staticmethod
+  @builder
+  def get(val: Attribute, res: DataType) -> 'Literal':
+    return Literal.build(attributes={"value": val}, result_types=[res])
+
+
+class Operator(Operation):
+  ...
+
+
+@irdl_op_definition
+class PandasTable(Operator):
+  """
+  Defines a table with name `table_name`.
+
+  Example:
+
+  '''
+  %0 : rel_ssa.bag<[!rel_ssa.int32]> = rel_ssa.pandas_table() ["table_name" = "t"]
+  '''
+  """
+  name = "rel_ssa.pandas_table"
+
+  table_name = AttributeDef(StringAttr)
+  result = ResultDef(Bag)
+
+  @staticmethod
+  @builder
+  def get(name: str, result_type: Attribute) -> 'PandasTable':
+    return PandasTable.build(
+        attributes={"table_name": StringAttr.from_str(name)},
+        result_types=[result_type])
+
+
+@irdl_op_definition
+class Select(Operator):
+  """
+  Selects all tuples of `input` according to `predicates`.
```

Comment:
```suggestion
  Selects all tuples of `input` according to the Expressions `predicates`.
```

or similar... (make the connection between expressions and operators explicit in the doc).